teachertool/src/transforms/getRubricFromFileAsync.ts

```diff
@@ -0,0 +1,25 @@
+import { logDebug, logError } from "../services/loggingService";
+import { postNotification } from "./postNotification";
+import { makeNotification } from "../utils";
+import { loadRubricFromFileAsync } from "../services/fileSystemService";
+import { verifyRubricIntegrity } from "../state/helpers";
+import { Rubric } from "../types/rubric";
+
+export async function getRubricFromFileAsync(file: File, allowPartial: boolean): Promise<Rubric | undefined> {
+    let rubric = await loadRubricFromFileAsync(file);
+
+    if (rubric) {
+        logDebug(`Loading rubric '${rubric.name}' from file...`);
+
+        const rubricVerificationResult = verifyRubricIntegrity(rubric);
+
+        if (!rubricVerificationResult.valid) {
+            postNotification(makeNotification(lf("Rubric contains invalid criteria."), 2000));
```

Comment:
Probably remove the notifications in this file. They aren't very visible and modal shows an error anyway.